src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt

```diff
@@ -118,11 +118,9 @@ class TransportStopIndexReplicationAction @Inject constructor(transportService:
                     }
                 }
                 val replMetadata = replicationMetadataManager.getIndexReplicationMetadata(request.indexName)
-                // If paused , we need to make attempt to clear retention leases as Shard Tasks are non-existent
-                if (isPaused) {
-                    log.info("Index[${request.indexName}] is in paused state")
-                    attemptRemoveRetentionLease(replMetadata, request.indexName)
-                }
+                val remoteClient = client.getRemoteClusterClient(replMetadata.connectionName)
+                val retentionLeaseHelper = RemoteClusterRetentionLeaseHelper(clusterService.clusterName.value(), remoteClient)
+                retentionLeaseHelper.attemptRemoveRetentionLease(clusterService, replMetadata, request.indexName)
```

Comment:
we can remove `isPaused` altogether now. `validateStopReplicationRequest` will not return bool then  .

Reply:
+1

Reply:
Done